src/main/java/com/esri/samples/na/service_areas/ServiceAreaTaskController.java

```diff
@@ -0,0 +1,259 @@
+    polygonOverlay.getGraphics().clear();
+    barrierOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Solves Service Areas Task using the facilities and barriers that were added to the mapview.
+   * <p>
+   * All service areas that are return will be displayed to the mapview.
+   */
+  @FXML
+  private void showServiceAreas() {
+
+    //turn barrier button off and add any barriers to service area parameters
+    btnAddBarrier.setSelected(false);
+    List<PolylineBarrier> polylineBarriers = new ArrayList<>();
+    barrierOverlay.getGraphics().forEach(graphic -> {
+      polylineBarriers.add(new PolylineBarrier((Polyline) graphic.getGeometry()));
+    });
+    serviceAreaParameters.setPolylineBarriers(polylineBarriers);
+
+    // need at least one facility for the task to work
+    if (serviceAreaFacilities.size() > 0) {
+      // get rid of any previous service area graphics
+      polygonOverlay.getGraphics().clear();
+      serviceAreaParameters.setFacilities(serviceAreaFacilities);
+      // find service areas around facility using parameters that were set
+      ListenableFuture<ServiceAreaResult> result = task.solveServiceAreaAsync(serviceAreaParameters);
+      result.addDoneListener(() -> {
+        try {
+          // display all service areas for all service are facilities and display to mapview
+          List<Graphic> graphics = polygonOverlay.getGraphics();
+          ServiceAreaResult serviceAreaResult = result.get();
+          for (int i = 0; i < serviceAreaFacilities.size(); i++) {
+            List<ServiceAreaPolygon> polygons = serviceAreaResult.getResultPolygons(i);
+            for (int j = 0; j < polygons.size(); j++) {
+              graphics.add(new Graphic(polygons.get(j).getGeometry(), fillSymbols.get(j % 3)));
+            }
+          }
+        } catch (ExecutionException | InterruptedException e) {
+          if (e.getMessage().contains("Unable to complete operation")) {
+            showErrorMessage("Facility not within SanFrancisco area!");
+          } else {
+            e.printStackTrace();
+          }
+        }
+      });
+    } else {
+      showErrorMessage("Must have at least 1 Facility!");
+    }
+  }
+
+  /** 
+   * Shows error message to user if something went wrong with solving task.
+   * 
+   * @param message error message to show.
+   */
+  private void showErrorMessage(String message) {
+    Alert dialog = new Alert(AlertType.WARNING);
+    dialog.setHeaderText(null);
+    dialog.setTitle("Error");
+    dialog.setContentText(message);
+    dialog.showAndWait();
+  }
+
+  /**
+   * Creates objects for displaying graphics to mapview.
+   */
+  private void setDisplayValues() {
+    // setup colors for multiple service areas
+    SimpleLineSymbol outline = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF000000, 3.0f);
+    fillSymbols = new ArrayList<>();
+    fillSymbols.add(new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, 0x6600FF00, outline));

```

Comment:
I think there are suppose to be more than just one Service Area but I will take this out and updated it if the needs to be fixed.